astropy/conftest.py

```diff
@@ -31,9 +32,6 @@
     # The above checks whether we are running in a PyInstaller bundle.
     warnings.filterwarnings("ignore", "(?s).*MATPLOTLIBDATA.*", category=UserWarning)
```

Comment:
As mentioned in the error message from #10785, MATPLOTLIBDATA has been removed in matplotlib 3.3 (https://matplotlib.org/stable/api/prev_api_changes/api_changes_3.3.0.html#environment-variables) so I think filtering this warning is no more needed ?

Reply:
good point, follow up PR incoming